### PR TITLE
Properly suggest `E::assoc` when we encounter `E::Variant::assoc`

### PR DESCRIPTION
**Pull request by estebank**
Use the right span when encountering an enum variant followed by an associated item so we don't lose the associated item in the resulting code.
    
Do not suggest the thing twice, once as a removal of the associated item and a second time as a typo suggestion.

### CODE REVIEW

**Comment by compiler-errors** [comment]
busy

r? compiler

File: compiler/rustc_resolve/src/late/diagnostics.rs [hunk 1]
```diff
@@ -589,7 +597,16 @@ impl<'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
         span: Span,
         res: Option<Res>,
         base_error: &BaseError,
-    ) -> (bool, Vec<ImportSuggestion>) {
+    ) -> (bool, FxHashSet<String>, Vec<ImportSuggestion>) {
+        let span = match following_seg {
+            Some(_) if path[0].ident.span.eq_ctxt(path[path.len() - 1].ident.span) => {
```
Comment (Nadrieril) [inline compiler/rustc_resolve/src/late/diagnostics.rs:602]: Is the branch on `following_seq` necessary? Presumably the path-derived span would be the same as the input `span` but is that actually the case?

Comment (estebank) [reply compiler/rustc_resolve/src/late/diagnostics.rs:602]: [Macros](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=25367c71f98c1a1c4e6628d82696305a).

Comment (Nadrieril) [reply compiler/rustc_resolve/src/late/diagnostics.rs:602]: It's always macros

**Pull request merged**